src/static/bitbucket/utils/getWidgetSettings.ts

```diff
@@ -1,10 +1,25 @@
 import { WidgetSettings } from '../../../types';
+import { proxyRequestBare } from './RequestProxy';
+import useState from 'react-usestateref';
+import { useEffect } from 'react';
 
-export default function getWidgetSettings(): WidgetSettings {
-  return {
-    refreshInterval: 10000,
-    refreshOnlyWhenInViewport: false,
-    // @ts-ignore -- this is a global variable that may be set by the server
-    ...window.__WIDGET_SETTINGS__,
-  };
+const defaultSettings = {
```

Comment:
The defaults could stay on the server and we can make the UI dumb by just using whatever the endpoint returns.

Reply:
Yeah, though there is a brief moment at the start when this setting isn't available to the client. I didn't want to block the UI on this information being available (we're fine to use defaults in the meanwhile).